docs/content/concepts/partitions-schedules-sensors/partitioned-assets.mdx

```diff
@@ -0,0 +1,383 @@
+)
+
+
+@sensor(job=images_job)
+def image_sensor(context):
+    new_images = [
+        img_filename
+        for img_filename in os.listdir(os.getenv("MY_DIRECTORY"))
+        if not context.instance.has_dynamic_partition(
+            images_partitions_def.name, img_filename
+        )
+    ]
+
+    return SensorResult(
+        run_requests=[
+            RunRequest(partition_key=img_filename) for img_filename in new_images
+        ],
+        dynamic_partitions_requests=[
+            images_partitions_def.build_add_request(new_images)
+        ],
+    )
+```
+
+### Partitioned assets with partitioned I/O managers
+
+Asset functions can write data out to files, but they can also delegate the writing operation to an [I/O manager](/concepts/io-management/io-managers). Dagster's [built-in I/O managers](/concepts/io-management/io-managers#built-in-io-managers) support handling partitioned assets, but you can also [write your own I/O manager](/concepts/io-management/io-managers#handling-partitioned-assets) if you want additional customization.
+
+For example, this example demonstrates how to define an asset that relies on an I/O manager to store its output:
+
+```python file=/concepts/partitions_schedules_sensors/partitioned_asset_uses_io_manager.py
+import pandas as pd
+
+from dagster import DailyPartitionsDefinition, asset
+
+
+@asset(partitions_def=DailyPartitionsDefinition(start_date="2022-01-01"))
+def my_daily_partitioned_asset(context) -> pd.DataFrame:
+    partition_date_str = context.asset_partition_key_for_output()
+    return pd.read_csv(f"coolweatherwebsite.com/weather_obs&date={partition_date_str}")
+```
+
+If using the default I/O manager, materializing partition `2022-07-23` of this asset would store the output `DataFrame` in a pickle file at a path like `my_daily_partitioned_asset/2022-07-23`.
+
+---
+
+## Defining partition dependencies
+
+Partitioned assets can depend on other partitioned assets. In this case, each partition in the downstream asset will depend on a partition or multiple partitions in the upstream asset.
+
+### Default partition dependency rules
+
+A few rules govern default partition-to-partition dependencies:
+
+- When the upstream asset and downstream asset have the same <PyObject object="PartitionsDefinition" />, each partition in the downstream asset will depend on the same partition in the upstream asset.
+- When the upstream asset and downstream asset are both time window-partitioned, each partition in the downstream asset will depend on all partitions in the upstream asset that intersect its time window.
+
+  For example, if an asset with a <PyObject object="DailyPartitionsDefinition" /> depends on an asset with an <PyObject object="HourlyPartitionsDefinition" />, then partition `2022-04-12` of the daily asset would depend on 24 partitions of the hourly asset: `2022-04-12-00:00` through `2022-04-12-23:00`.
+
+### Overriding default dependency rules using PartitionMappings
+
+<Note>
```

Comment:
this is no longer true!! as of the most recent release you can use `AssetDep` to do partition mappings 

```python
from dagster import (
    AssetDep,
    DailyPartitionsDefinition,
    TimeWindowPartitionMapping,
    asset,
)

partitions_def = DailyPartitionsDefinition(start_date="2023-01-21")


@asset(partitions_def=partitions_def)
def events():
    ...


@asset(
    partitions_def=partitions_def, 
    deps=[AssetDep(events, partition_mapping=TimeWindowPartitionMapping(start_offset=-1, end_offset=-1))]

)
def yesterday_event_stats():
    ...
```

maybe we use this ^ example as the first example and then have a subsection called something like "For managed-loading dependencies" with the other example

Reply:
@jamiedemaria Would you happen to have a test for this snippet I could plug into our examples?